snackpub/src/NetworkUtils.ts

```diff
@@ -0,0 +1,12 @@
+import type { IncomingHttpHeaders, IncomingMessage } from 'http';
+
+type ValueOf<T> = T[keyof T];
+
+export function getRemoteAddress(req: IncomingMessage): string | undefined {
+  return getFirstHeaderValue(req.headers['x-forwarded-for']) ?? req.socket.remoteAddress;
```

Comment:
Look at server/www/src/middleware/ip.ts. I think it would be a good idea to proactively add support for cf-connecting-ip if we were to add Cloudflare proxies. (I'm just worried that we might add Cloudflare proxies and then break the rate limiter because all of the requests will appear to come from a small set of the same IP addresses owned by the proxy servers.)